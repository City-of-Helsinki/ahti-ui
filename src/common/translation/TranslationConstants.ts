export const SUPPORTED_LANGUAGES: Record<string, string> = {
  FI: 'fi',
  EN: 'en'
};
