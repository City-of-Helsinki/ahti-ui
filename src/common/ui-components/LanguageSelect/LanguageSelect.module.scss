@import '../../../styles/partials/variables';
@import '../../../styles/partials/mixins';

.languageButton {
  @include buttonReset;
  margin: $margin/4;
}

.languageButtonEnabled {
  font-weight: $font-width-thick;
}
