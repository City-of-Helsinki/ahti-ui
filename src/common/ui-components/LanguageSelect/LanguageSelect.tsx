import React from 'react';
import { useTranslation } from 'react-i18next';
import classnames from 'classnames/bind';
import styles from './LanguageSelect.module.scss';

const cx = classnames.bind(styles);

export interface LanguageSelectProps {
  readonly supportedLanguages: string[];
  readonly className?: string;
}

const LanguageSelect: React.FC<LanguageSelectProps> = ({
  supportedLanguages,
  className
}) => {
  const { i18n } = useTranslation();

  return (
    <div className={className}>
      {supportedLanguages.map((language: string, id: number) => {
        return (
          <button
            key={id}
            onClick={() => i18n.changeLanguage(language)}
            className={cx({
              languageButton: true,
              languageButtonEnabled: language === i18n.language
            })}
            lang={language}
          >
            {language}
          </button>
        );
      })}
    </div>
  );
};

export default LanguageSelect;
