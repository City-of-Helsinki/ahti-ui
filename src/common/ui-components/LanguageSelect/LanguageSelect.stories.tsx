import React from 'react';
import LanguageSelect from './LanguageSelect';

import { SUPPORTED_LANGUAGES } from '../../translation/TranslationConstants';

export default {
  title: 'LanguageSelect',
  component: LanguageSelect
};

export const SupportedLanguages = () => (
  <LanguageSelect supportedLanguages={Object.values(SUPPORTED_LANGUAGES)} />
);

SupportedLanguages.story = {
  name: 'Supported languages'
};
