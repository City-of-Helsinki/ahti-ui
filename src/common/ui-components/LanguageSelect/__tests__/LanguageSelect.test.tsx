import React from 'react';
import { shallow } from 'enzyme';
import LanguageSelect from '../LanguageSelect';
import { SUPPORTED_LANGUAGES } from '../../../translation/TranslationConstants';
import i18next from 'i18next';

describe('LanguageSelect', () => {
  const supportedLanguages = Object.values(SUPPORTED_LANGUAGES);
  const wrapper = shallow(
    <LanguageSelect supportedLanguages={supportedLanguages} />
  );

  it('renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders all supported languages as buttons', () => {
    expect(wrapper.find('.languageButton')).toHaveLength(
      supportedLanguages.length
    );
  });

  it('changes language on click', () => {
    const mockOnLanguageChange = jest.fn();
    i18next.on('languageChanged', mockOnLanguageChange);
    expect(mockOnLanguageChange.mock.calls.length).toBe(0);
    wrapper
      .find('.languageButton')
      .not('.languageButtonEnabled')
      .simulate('click');
    expect(mockOnLanguageChange.mock.calls.length).toBe(1);
  });
});
