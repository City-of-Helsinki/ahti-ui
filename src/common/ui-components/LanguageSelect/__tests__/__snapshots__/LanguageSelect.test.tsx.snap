// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelect renders the component 1`] = `"<div><button class=\\"languageButton\\" lang=\\"fi\\">fi</button><button class=\\"languageButton languageButtonEnabled\\" lang=\\"en\\">en</button></div>"`;
